other_data_collections/2015ApJ...812...60B/Add_Biteau.py

```diff
@@ -0,0 +1,190 @@
+# This script reads in Biteau's catalog stored in 
+# BiteauWilliams2015_AllData_ASDC_v2016_12_20.ecsv,
+# it creates the corresponding ecsv-sed files and stores
+# them in /input/data
+
+from astropy.table import Table
+from astropy.io import ascii
+from astropy.constants import h,e
+from pathlib import Path
+import astropy.units as u
+import numpy as nump
+import os
+import glob
+
+class Biteau:
+    fBiteauFile = './BiteauWilliams2015_AllData_ASDC_v2016_12_20.ecsv'
+    fBiteauSourceIDFile = './Biteau_Sources.txt'
+    fBiteauExperimentFile = './Biteau_Experiments.txt'
+    fInputDataPath = './../../input/data/'
+
+    def __init__(self):
+        self.fBiteauCatalog = self.Load_Catalog(self.fBiteauFile)
+        self.fBiteauSources = self.Load_SourceIDs(self.fBiteauSourceIDFile)
+        self.fBiteauExperiments = self.Load_Experiments(self.fBiteauExperimentFile)
+
+# Load_Catalog reads Biteau's catalog
+    def Load_Catalog(self, filename):
+        print('Reading {}'.format(filename))
+        biteau_catalog = Table.read(filename, format = 'ascii.ecsv', delimiter='|')
+        return biteau_catalog
+
+# Load_SourceIDs read Biteau_Sources.txt in which all sources which are mentioned in
+# Biteau's catalog and the corresponding source_id are stored
+    def Load_SourceIDs(self, filename):
+        biteau_sources = dict()
+        print('Reading {}'.format(filename))
+        data = ascii.read(filename)
+        for i in range(0, len(data)):
+            biteau_sources[data[i][0]] = data[i][1]
+        return biteau_sources
+
+# Load_Experiments read Biteau_Experiments.txt in which all sources which are mentioned in
+# Biteau's catalog and the corresponding telescope from /input/schemas are stored
+    def Load_Experiments(self, filename):
+        biteau_experiments = dict()
+        print('Reading {}'.format(filename))
+        data = ascii.read(filename)
+        for i in range(0, len(data)):
+            biteau_experiments[data[i][0]] = data[i][1]
+        return biteau_experiments
+
+# Rename_Experiment renames all source names in Biteau's catalog into the corresponding name
+# in /input/schemas
+    def Rename_Experiment(self, experiment):
+        return self.fBiteauExperiments[experiment]
+
+# Calculate_Energy transforms the 'freq'-column in Biteau's catalog into an energy-column 
+    def Calculate_Energy(self, frequency):
+        energy = (h*frequency*1/u.second)/e.si * 1E-9* u.teraelectronvolt/u.joule*u.coulomb
+        return energy
+
+# Calculate_dnde transforms the 'e2dnde'-column in Biteau's catalog into an dnde-column
+    def Calculate_dnde(self, energy, e2dnde):
+        dnde = 1/((energy*u.teraelectronvolt)**2)*e2dnde*1/e.si*1E-16*u.erg*1/((u.centimeter)**2)/u.second*u.teraelectronvolt/u.erg
+        return dnde
+
+# Create_Subtables splits up Biteau's catalog into the single data sets and stores these sets 
+# in a list which contains astropy tables
+    def Create_Subtables(self):
```

Comment:
I think your code to make the subtables is much longer / complex than it needs to be.
Suggest to make a mask with the row subset selection and then index into the table to get the sub-table with that mask, like I did here: https://nbviewer.jupyter.org/gist/cdeil/2942d2f0241d7ff8d25db8008e0fda03
Should just be ~ 5 lines, no?